streams/src/main/java/org/apache/kafka/streams/state/Stores.java

```diff
@@ -215,6 +218,16 @@ public static WindowBytesStoreSupplier persistentWindowStore(final String name,
         if (segmentInterval < 60_000) {
             throw new IllegalArgumentException("segmentInterval must be at least one minute");
         }
+        if (windowSize > retentionPeriod) {
+            throw new IllegalArgumentException("The retention period of the window store "
+                                                   + name + " must be no smaller than its window size. Got size=["
+                                                   + windowSize + "], retention=[" + retentionPeriod + "]");
+        }
+        if (windowSize > segmentInterval) {
+            throw new IllegalArgumentException("The segment interval of the window store "
+                                                   + name + " must be no smaller than its window size. Got size=["
+                                                   + windowSize + "], segmentInterval=[" + segmentInterval + "]");
+        }
```

Comment:
It seems like we should also check this as well. Or is it ok for windows to be bigger than segments?

Reply:
I think it might be ok for `windowSize` to be greater than `segmentInterval` as the window will comprise multiple segments, but I could be wrong about this.

\cc @guozhangwang  @mjsax